vercel.json

```diff
@@ -50,243 +50,63 @@
       "has": [
         {
           "type": "query",
           "key": "lang",
           "value": "php"
         }
       ],
-      "destination": "/dev-guide/php/foundations:path*",
+      "destination": "/dev-guide/php:path*",
       "permanent": false
     },
     {
-      "source": "/application-development/features:path*",
-      "has": [
-        {
-          "type": "query",
-          "key": "lang",
-          "value": "php"
-        }
-      ],
-      "destination": "/dev-guide/php/features:path*",
-      "permanent": false
-    },
-    {
-      "source": "/application-development/observability:path*",
-      "has": [
-        {
-          "type": "query",
-          "key": "lang",
-          "value": "php"
-        }
-      ],
-      "destination": "/dev-guide/php/observability:path*",
-      "permanent": false
-    },
-    {
-      "source": "/application-development/testing:path*",
-      "has": [
-        {
-          "type": "query",
-          "key": "lang",
-          "value": "php"
-        }
-      ],
-      "destination": "/dev-guide/php/testing:path*",
-      "permanent": false
-    },
-    {
-      "source": "/application-development/foundations:path*",
-      "has": [
-        {
-          "type": "query",
-          "key": "lang",
-          "value": "python"
-        }
-      ],
-      "destination": "/dev-guide/python/foundations:path*",
-      "permanent": false
-    },
-    {
-      "source": "/application-development/features:path*",
-      "has": [
-        {
-          "type": "query",
-          "key": "lang",
-          "value": "python"
-        }
-      ],
-      "destination": "/dev-guide/python/features:path*",
-      "permanent": false
-    },
-    {
-      "source": "/application-development/observability:path*",
-      "has": [
-        {
-          "type": "query",
-          "key": "lang",
-          "value": "python"
-        }
-      ],
-      "destination": "/dev-guide/python/observability:path*",
-      "permanent": false
-    },
-    {
-      "source": "/application-development/testing:path*",
+      "source": "/application-development:path*",
       "has": [
         {
           "type": "query",
           "key": "lang",
           "value": "python"
         }
       ],
-      "destination": "/dev-guide/python/testing:path*",
-      "permanent": false
-    },
-    {
-      "source": "/application-development/foundations:path*",
-      "has": [
-        {
-          "type": "query",
-          "key": "lang",
-          "value": "typescript"
-        }
-      ],
-      "destination": "/dev-guide/typescript/foundations:path*",
-      "permanent": false
-    },
-    {
-      "source": "/application-development/features:path*",
-      "has": [
-        {
-          "type": "query",
-          "key": "lang",
-          "value": "typescript"
-        }
-      ],
-      "destination": "/dev-guide/typescript/features:path*",
+      "destination": "/dev-guide/python:path*",
```

Comment:
deployment preview:

/application-development/testing#replay?lang=python

takes me to:

/dev-guide#replay?lang=python

as if it's not matching this rule

Reply:
Pretty sure that is because the query param never worked when it was after the anchor?
For example, if I recall, it would have had to have been `/application-development/testing?lang=python#replay` -
Otherwise the query param would have been ignored.
You can go to this docs page and test it out: https://docusaurus.io/docs/markdown-features/tabs#query-string
Just move the query param to the end - and you will see it doesn't work.

Reply:
Ah right, thanks. Here's anchor last, same behavior:


```
visiting:
https://docs.temporal.io/application-development/testing?lang=python#replay
goes to:
https://docs.temporal.io/dev-guide?lang=python#replay
desired:
https://docs.temporal.io/dev-guide/python/testing#replay
```

Reply:
Ya, I swear these worked when I tested, but I may have been looking at the wrong preview.
In any case: https://github.com/temporalio/documentation/pull/2044
This should do all of it now.